source/configuration_manual/welcome_plugin.rst

```diff
@@ -10,6 +10,12 @@ Call a script when the user logs in for the first time. This is specifically
 done when the INBOX is (auto)created. The scripts are called similarly to
 :ref:`quota_configuration_warning_scripts`.
 
+.. note:: The welcome plugin relies on creation or autocreation of INBOX.
+          If the mail_location is maildir there can be situations where the
+          welcome script is not triggered. That is the case when the user
+          home dir is created by a IMAP command that does not interact with
```

Comment:
Isn't this fixed for Maildir by your changes?

Reply:
Maybe this could be a .. changedsince, and say this was fixed for everything except Maildir?

Reply:
And I suppose it's Maildir only when INBOX = Maildir/ directory, so e.g. using mail_location = maildir:~/Maildir:INBOX=~/Maildir/.INBOX works around this?

Reply:
Yes right, only happens without the non-default INBOX location. I added that to the description here.